### PR TITLE
Preventing index field type polling running twice for the same index.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

Prior to this change, when polling the field types of an index set took longer than 1 second, a second polling job was submitted to the executor, running in parallel. If it took longer than 2s, a third one is started, and so on.

In order to prevent this, a very simple locking mechanism is implemented, where a running polling job adds the current index set's id to a concurrent hash set, removing it when finished (or failing).

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/indexer/fieldtypes/IndexFieldTypePollerPeriodical.java [hunk 1]
```diff
@@ -164,20 +168,30 @@ private void poll(Collection<IndexSetConfig> indexSetConfigs) {
...
                 .filter(IndexSetConfig::isWritable)
-                .filter(config -> {
+                .forEach(config -> {
                     final Instant previousPoll = lastPoll.getOrDefault(config.id(), Instant.MIN);
                     final Instant nextPoll = previousPoll.plusSeconds(
                             config.fieldTypeRefreshInterval().getStandardSeconds());
-                    return !Instant.now().isBefore(nextPoll);
-                })
-                .forEach(this::poll);
+                    if (!Instant.now().isBefore(nextPoll)) {
```
Comment (luk-kaminski) [inline graylog2-server/src/main/java/org/graylog2/indexer/fieldtypes/IndexFieldTypePollerPeriodical.java:175]: Won't `Instant.now().isAfter()` be slightly more readable?

Comment (dennisoelkers) [reply graylog2-server/src/main/java/org/graylog2/indexer/fieldtypes/IndexFieldTypePollerPeriodical.java:175]: I had the same thought. I think the intention was that it should also start when when `now == nextPoll`.

File: graylog2-server/src/test/java/org/graylog2/indexer/fieldtypes/IndexFieldTypePollerPeriodicalTest.java [hunk 2]
```diff
@@ -62,14 +81,55 @@ void setUp() throws Exception {
...
+        final ReentrantLock lock = new ReentrantLock();
+        lock.lock();
+        when(indexFieldTypePoller.pollIndex(anyString(), anyString()))
+                .thenAnswer((Answer<Optional<IndexFieldTypesDTO>>) invocationOnMock -> {
+                    lock.lock();
+                    return Optional.empty();
+                });
+
+        periodical.doRun();
+        periodical.doRun();
```
Comment (luk-kaminski) [inline graylog2-server/src/test/java/org/graylog2/indexer/fieldtypes/IndexFieldTypePollerPeriodicalTest.java:130]: Those 2 invocations does not seem to be run in different threads, they are run sequentially in the test's thread.
Therefore, in my opinion, lock usage has completely no impact on the test behavior.

Comment (dennisoelkers) [reply graylog2-server/src/test/java/org/graylog2/indexer/fieldtypes/IndexFieldTypePollerPeriodicalTest.java:130]: Good catch! I overlooked that we are using a single thread executor. Fixed that!

**Review by luk-kaminski — APPROVED** [review]
Great job, thanks a lot!

**Pull request merged**